Source/JavaScriptCore/runtime/JSONObject.cpp

```diff
@@ -713,10 +717,84 @@ void FastStringifier::logOutcome(String&& outcome)
 
 #endif
 
+inline unsigned FastStringifier::useableBufferSize(size_t availableBufferSize)
+{
+    // FastStringifier relies on remaining useable capacity in m_buffer to limit recursion
```

Comment:
Are you sure this super-long comment is the best way to explain things? Can we tell the story in fewer words?

Reply:
There are many details and assumptions (not obvious from the code here) that provide context for the reasoning behind the estimation algorithm.  I think there is value in documenting clearly what these are in case we want to tweak them in the future.  Incidentally, there was also a bug in the comment, which I've now fixed.  I've also reduced a bit of verbosity here and there.